timesketch/lib/utils.py

```diff
@@ -83,97 +90,99 @@ def _scrub_special_tags(dict_obj):
-
     # Ensures delimiter is a string.
     if not isinstance(delimiter, six.text_type):
         delimiter = codecs.decode(delimiter, 'utf8')
 
-    # Due to issues with python2.
-    if six.PY2:
-        delimiter = str(delimiter)
-
-    reader = csv.DictReader(file_handle, delimiter=delimiter)
-    csv_header = reader.fieldnames
-    missing_fields = []
-    # Validate the CSV header
-    for field in mandatory_fields:
-        if field not in csv_header:
-            missing_fields.append(field)
-    if missing_fields:
-        raise RuntimeError(
-            'Missing fields in CSV header: {0:s}'.format(
-                ','.join(missing_fields)))
+    header_reader = pandas.read_csv(file_handle, sep=delimiter, nrows=0)
+    _validate_csv_fields(mandatory_fields, header_reader)
     try:
-        for row in reader:
-            # There is a condition in which the CSV reader can read a
-            # single lines as multiple, causing issues with importing.
-            # TODO: Swap the CSV library for the user of pandas.
-            if not row:
-                continue
-            if not row['datetime']:
+        reader = pandas.read_csv(file_handle, sep=delimiter,
+                                 chunksize=DEFAULT_CHUNK_SIZE)
+        for chunk in reader:
+            skipped_rows = chunk[chunk['datetime'].isnull()]
+            for row in skipped_rows.iterrows():
                 logger.warning(
                     'Row missing a datetime object, skipping [{0:s}]'.format(
                         ','.join([str(x).replace(
                             '\n', '').strip() for x in row.values()])))
-                continue
-            try:
-                # normalize datetime to ISO 8601 format if it's not the case.
-                parsed_datetime = parser.parse(row['datetime'])
-                row['datetime'] = parsed_datetime.isoformat()
-
-                normalized_timestamp = int(
-                    time.mktime(parsed_datetime.utctimetuple()) * 1000000)
-                normalized_timestamp += parsed_datetime.microsecond
-                row['timestamp'] = str(normalized_timestamp)
-                if 'tag' in row:
-                    row['tag'] = [x for x in _parse_tag_field(row['tag']) if x]
 
-                _scrub_special_tags(row)
-            except ValueError:
-                continue
+            # Normalize datetime to ISO 8601 format if it's not the case.
+            chunk['datetime'] = pandas.to_datetime(chunk['datetime'])
 
-            yield row
-    except csv.Error as e:
+            chunk['timestamp'] = chunk['datetime'].apply(_get_unix_time)
+            chunk['datetime'] = chunk['datetime'].apply(pandas.Timestamp
+                                                        .isoformat).astype(str)
+
+            if 'tag' in chunk:
+                chunk['tag'] = chunk['tag'].apply(
+                    lambda tag: [x for x in _parse_tag_field(tag) if x]
+                    if tag else None)
```

Comment:
I feel like tiuo don't need to do this, since _parse_tag_field accepts a list, so are you sure you can't just do

```
chunk['tag'] = chunk['tag'].apply(_parse_tag_field)
```